File: Java/UpperTriangularMatrix.java
import java.util.*;

// this program converts a matrix to an upper triangular matrix

public class UpperTriangularMatrix {
    public static void printMatrix(int[][] arr) { 
        int m = arr.length;
        int n = arr[0].length;
        for (int i = 0; i < m; i++) { 
            for (int j = 0; j < n; j++) {
                System.out.print(arr[i][j] + " "); 
            }    
            System.out.println(); 
        } 
    }     

    public static void upperTriangularMatrix(int arr[][]) { 
        int m = arr.length; 
        int n = arr[0].length;         
        if (m != n) { 
            System.out.println("Matrix entered should be a Square Matrix");
            System.out.println("Try Again...");
            return; 
        } else { 
            for (int i = 0; i < m; i++) { 
                for (int j = 0; j < n; j++) { 
                    if (i > j) { 
                        arr[i][j] = 0; 
                    } 
                } 
            }   
            System.out.println("Upper Triangular Matrix is : ");
            printMatrix(arr); 
        } 
    } 

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter the number of rows: ");
        int m = sc.nextInt();

        System.out.println("Enter the number of columns: ");
        int n = sc.nextInt();

        System.out.println("Enter the matrix elements: ");
        int arr[][] = new int[m][n];

        for(int i = 0; i < m; i++) {
            for(int j = 0; j < n; j++) {
                arr[i][j]=sc.nextInt();
            }
        }
    
        System.out.println("Original Matrix is : "); 
        printMatrix(arr);         
      
        upperTriangularMatrix(arr); 
    } 
}

File: C++/UpperTriangularMatrix.cpp
#include <iostream>
using namespace std;

// this program converts a matrix to an upper triangular matrix

void printMatrix(int** arr, int m, int n) {
    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            cout << arr[i][j] << " ";
        }
        cout << endl;
    }
}

void upperTriangularMatrix(int** arr, int m, int n) {
    if (m != n) {
        cout << "Matrix entered should be a Square Matrix" << endl;
        cout << "Try Again..." << endl;
        return;
    } else {
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (i > j) {
                    arr[i][j] = 0;
                }
            }
        }
        cout << "Upper Triangular Matrix is : " << endl;
        printMatrix(arr, m, n);
    }
}

int main() {
    int m, n;

    cout << "Enter the number of rows: " << endl;
    cin >> m;

    cout << "Enter the number of columns: " << endl;
    cin >> n;

    // Allocate 2D array dynamically
    int** arr = new int*[m];
    for (int i = 0; i < m; i++) {
        arr[i] = new int[n];
    }

    cout << "Enter the matrix elements: " << endl;
    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            cin >> arr[i][j];
        }
    }

    cout << "Original Matrix is : " << endl;
    printMatrix(arr, m, n);

    upperTriangularMatrix(arr, m, n);

    // Free allocated memory
    for (int i = 0; i < m; i++) {
        delete[] arr[i];
    }
    delete[] arr;

    return 0;
}